### DISCUSSION TITLE
Head's Up:  Plex has started decommissioning Plug-Ins and Agents

### DISCUSSION THREAD

**Discussion by CodyBerenson** [CodyBerenson/PGMA-Modernized#315] [General]
Especially if you are an Nvidia Shield PMS user, please see this article:

https://forums.plex.tv/t/important-information-for-users-running-plex-media-server-on-nvidia-shield-devices/883484/1

This has been a long time coming, but looks like plug-ins and agents are officially dying soon. For now, its only Shield PMS.  Per the article, Plex is committed to providing an alternative framework to developers before killing Python based scraper agents (like PGMA).  

For those of you who have been with us for a while, you may recall that Plex killed all of our agents several PMS releases ago, and folks had to roll back to a previous PMS version (and stay there) until Plex fixed it.  

We don't know what the alternative looks like and whether its viable for us or not, nor do we know when.  Let's all say a little prayer to the porn god daddies and their flocks of twinks that we can continue to enjoy our special time and international access to our individual little corners of curated naughtiness.   I shudder to think of the countless hours of time freed up each week if I'm not renaming and indexing content.

**Comment by fourstix** [comment]
Bummer.  But there have been some vague hints this was coming in the past.  I haven't heard anything about the "new framework" beyond that Plex plans to announce one someday.

**Comment by JPH71** [comment]
We should start exploring other options.... aside from plex.... as an option

**Comment by joca879** [comment]
I've heard about another media server similar to Plex, Emby. But I never tried it and don't have knowledge about it

**Comment by fourstix** [comment]
There's also another one called JellyFin which I believe is an open source version
of Plex

**Comment by fourstix** [comment]
Plex announced in their forum that they plan to support an Open Source API for Third Party Agents, NFO support and other features. 

The r/Plex reddit forum has a good summary with a link to the original and some interesting discussion on this.

https://www.reddit.com/r/PleX/comments/1feky0s/open_sourced_plex_api_nfo_support_custom_agents/

**Comment by JPH71** [comment]
Nice one!

**Comment by CodyBerenson** [comment]
Welcome relief.  That puts this PGMA project on the path to closing shop and going out of business.  Retirement, here I come.

**Comment by jfreiman** [comment]
[https://forums.plex.tv/t/fireside-follow-up-metadata-agenda-nfo-support-and-some-of-our-future-work-on-pms/887946]

I would happily donate $100 to PGMA (and more going forward) if you take on the massive task of rewriting PGMA for Plex's newly announced API.

I truly love and also appreciate everyone involved in making this a world class accessory to the Plex ecosystem. 

PS. I'm the past month I have looked high and low for a donation link.... If their is one that I missed, please post it so I can show my appreciation.
